apps/api/src/campaign/campaign.controller.ts

```diff
@@ -119,8 +119,15 @@ export class CampaignController {
     roles: [RealmViewSupporters.role, ViewSupporters.role],
     mode: RoleMatchingMode.ANY,
   })
-  async update(@Param('id') id: string, @Body() updateCampaignDto: UpdateCampaignDto) {
-    return this.campaignService.update(id, updateCampaignDto)
+  async update(@Param('id') id: string, @Body() updateCampaignDto: UpdateCampaignDto, @AuthenticatedUser() user: KeycloakTokenParsed) {
+    const campaign = await this.campaignService.getCampaignByIdWithPersonIds(id)
+    if (user.sub === campaign?.beneficiary.person?.keycloakId ||
```

Comment:
since this is critical functionality for permisions check, please add tests to ensure all conditions are covered, so that we're protected from breaking the logic in future

Reply:
I totally forgot about the tests.I'll implement the tests later tommorow